### PR TITLE
feat(network): add firewall black list ips to banning when connecting to peer

### PR DESCRIPTION
**Pull request by Ja7ad**
# Description

This PR defines a blacklist that contains IP addresses that should be blocked. When a node attempts to connect, its IP address is checked against the blacklist. If it matches an entry, the connection is rejected.

### CODE REVIEW

File: sync/firewall/config.go [hunk 1]
```diff
@@ -1,16 +1,42 @@
...
+//go:embed black_list.json
+var _defaultBlackListIPs []byte
+
 type Config struct {
-	Enabled bool `toml:"enable"`
+	Enabled      bool     `toml:"enable"`
+	BlackListIPs []string `toml:"blacklist_ips"`
+}
+
+type defaultBlackListIPs struct {
```
Comment (b00f) [inline sync/firewall/config.go:16]: Why we have defaulted here?

File: sync/firewall/black_list.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "ips": [
+    "115.193.157.138"
```
Comment (b00f) [inline sync/firewall/black_list.json:3]: Looks redundant to me

File: sync/firewall/firewall_test.go [hunk 3]
```diff
@@ -78,6 +84,16 @@ func TestInvalidBundlesCounter(t *testing.T) {
 	assert.Equal(t, p.InvalidBundles, 3)
 }
 
+func TestBlackListIPs(t *testing.T) {
+	td := setup(t)
+
+	blackListIP := "10.10.10.10"
```
Comment (b00f) [inline sync/firewall/firewall_test.go:90]: Remote address is in format of libp2p address. Update test

**Review by b00f — CHANGES_REQUESTED** [review]

File: sync/firewall/config.go (hunk 1 above)
Comment (Ja7ad) [reply sync/firewall/config.go:16]: This Default append to configuration list, some time we find faulty similar now we want banned on patch version.

Same now.

**Review by b00f — APPROVED** [review]

**Pull request merged**